<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Performing an example analysis using singularity - GRADitude</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Performing an example analysis using singularity";
    var mkdocs_page_input_path = "perform_an_example_analysis_using_docker.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> GRADitude</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../installation/">Installation</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../subcommands/">GRADitude's subcommands</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../perform_an_example_analysis/">Performing an example analysis</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Performing an example analysis using singularity</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#docker-containers-and-singularity">Docker containers and singularity</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../license/">License</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">GRADitude</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Performing an example analysis using singularity</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="tutorial-of-graditude-using-singularity">Tutorial of GRADitude using Singularity</h1>
<p>This tutorial will guide you through a test case to show how to 
perform a GRAD-seq analysis from the pre-processing steps to the
downstream analysis that will be done using GRADitude.
We recommend to execute this analysis in a server or in a virtual machine. The pre-processing 
and mapping steps require a lot of computational time and power.</p>
<p>In this tutorial we will try to show how to use all GRADitude's subcommands
in order to have an overview of all possible complexes.</p>
<p>In order to perform all these steps we need to have many tools and dependencies in 
your pc. To make all simpler and more reproducible we suggest following the example provided below.
All these steps can be then easily adjusted to analyze different 
GRAD-seq experiments.</p>
<h2 id="docker-containers-and-singularity">Docker containers and singularity</h2>
<p>Another way to distribute packages is using Docker. Docker is a service that deliver software
in so called containers. All containers have their own libraries and packages.</p>
<p>A Docker container has inside all dependencies and applications. This increase
the reproducibility of a specific analysis.
In this example all the tools required for the analysis are inside the image that will be generated.</p>
<p>In order to avoid root permission when running Docker, 
Singularity can be used.
For this reason, the example analysis provided, is done using singularity.</p>
<p>You can download singularity in <a href="https://sylabs.io/guides/3.0/user-guide/installation.html#">here</a></p>
<p>Once you have singularity in your computer you can run inside your project directory this command:</p>
<pre><code class="language-bash">singularity build grad-seq-analysis.sif docker://silviadigiorgio/grad-seq-analysis-ecoli
</code></pre>
<p>This will create an image that can be used to perform the analysis.
Before starting the analysis please create the folders input and output in the main directory.</p>
<p>This can be also done via shell using the bash command mkdir.</p>
<pre><code class="language-bash">mkdir input output
</code></pre>
<p>We recommend to create subfolders inside the input main folder</p>
<pre><code class="language-bash">cd input &amp;&amp;
mkdir data_folder raw_reads
</code></pre>
<p>At the end you will have inside the analysis folder the created image and the two folders</p>
<pre><code class="language-bash">.
├── grad-seq-analysis.sif
├── input
│   ├── data_folder
│   └── raw_reads
└── output

</code></pre>
<p>If you are not familiar with Bioinformatics please be aware that some commands we will execute 
they will take some time. So we recommend to run everything in a server and with <a href="https://github.com/tmux/tmux/wiki">tmux</a></p>
<p>To perform our test case, the raw data have to be retrieved from <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP268299">NCBI</a>.</p>
<p>In order to perform this step the user needs to use the fastq-dump function
provided by the SRA-toolkit that can be downloaded <a href="https://anaconda.org/bioconda/sra-tools">here</a></p>
<p>We suggest to download the raw data inside the input folder it has been previously created </p>
<pre><code class="language-bash">
cd input/raw_reads &amp;&amp;
    fastq-dump --bzip2 SRR12067299
    fastq-dump --bzip2 SRR12067300
    fastq-dump --bzip2 SRR12067301
    fastq-dump --bzip2 SRR12067302
    fastq-dump --bzip2 SRR12067303
    fastq-dump --bzip2 SRR12067304
    fastq-dump --bzip2 SRR12067305
    fastq-dump --bzip2 SRR12067306
    fastq-dump --bzip2 SRR12067307
    fastq-dump --bzip2 SRR12067308
    fastq-dump --bzip2 SRR12067309
    fastq-dump --bzip2 SRR12067310
    fastq-dump --bzip2 SRR12067311
    fastq-dump --bzip2 SRR12067312
    fastq-dump --bzip2 SRR12067313
    fastq-dump --bzip2 SRR12067314
    fastq-dump --bzip2 SRR12067315
    fastq-dump --bzip2 SRR12067316
    fastq-dump --bzip2 SRR12067317
    fastq-dump --bzip2 SRR12067318 
    fastq-dump --bzip2 SRR12067319 
    fastq-dump --bzip2 SRR12067320

</code></pre>
<p>or to automate the process you can run the same command using a for loop</p>
<pre><code class="language-bash">for i in            \
    SRR12067299     \
    SRR12067300     \
    SRR12067301     \
    SRR12067302     \
    SRR12067303     \
    SRR12067304     \
    SRR12067305     \
    SRR12067306     \
    SRR12067307     \
    SRR12067308     \
    SRR12067309     \
    SRR12067310     \
    SRR12067311     \
    SRR12067312     \
    SRR12067313     \
    SRR12067314     \
    SRR12067315     \
    SRR12067316     \
    SRR12067317     \
    SRR12067318     \
    SRR12067319     \
    SRR12067320     \
; do 
    fastq-dump --bzip2 $i; 
  done
</code></pre>
<p>After downloading the data I recommend to change the name of the files according the specific 
fraction like suggested here:</p>
<pre><code class="language-bash">cd input/raw_reads &amp;&amp;
mv SRR12067299.fastq.bz2 Grad-00L.fastq.bz2 &amp;&amp;
mv SRR12067300.fastq.bz2 Grad-01.fastq.bz2 &amp;&amp;
mv SRR12067301.fastq.bz2 Grad-02.fastq.bz2 &amp;&amp;  
mv SRR12067302.fastq.bz2 Grad-03.fastq.bz2 &amp;&amp;   
mv SRR12067303.fastq.bz2 Grad-04.fastq.bz2 &amp;&amp;   
mv SRR12067304.fastq.bz2 Grad-05.fastq.bz2 &amp;&amp;   
mv SRR12067305.fastq.bz2 Grad-06.fastq.bz2 &amp;&amp;   
mv SRR12067306.fastq.bz2 Grad-07.fastq.bz2 &amp;&amp;    
mv SRR12067307.fastq.bz2 Grad-08.fastq.bz2 &amp;&amp;   
mv SRR12067308.fastq.bz2 Grad-09.fastq.bz2 &amp;&amp;   
mv SRR12067309.fastq.bz2 Grad-10.fastq.bz2 &amp;&amp;   
mv SRR12067310.fastq.bz2 Grad-11.fastq.bz2 &amp;&amp;   
mv SRR12067311.fastq.bz2 Grad-12.fastq.bz2 &amp;&amp;    
mv SRR12067312.fastq.bz2 Grad-13.fastq.bz2 &amp;&amp;    
mv SRR12067313.fastq.bz2 Grad-14.fastq.bz2 &amp;&amp;   
mv SRR12067314.fastq.bz2 Grad-15.fastq.bz2 &amp;&amp;    
mv SRR12067315.fastq.bz2 Grad-16.fastq.bz2 &amp;&amp;   
mv SRR12067316.fastq.bz2 Grad-17.fastq.bz2 &amp;&amp;   
mv SRR12067317.fastq.bz2 Grad-18.fastq.bz2 &amp;&amp;    
mv SRR12067318.fastq.bz2 Grad-19.fastq.bz2 &amp;&amp;    
mv SRR12067319.fastq.bz2 Grad-20.fastq.bz2 &amp;&amp;   
mv SRR12067320.fastq.bz2 Grad-21P.fastq.bz2    
</code></pre>
<p>You can download the reference genome that will be used later on from the NCBI using this command</p>
<pre><code class="language-bash">
cd input/data_folder &amp;&amp;
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz &amp;&amp;
gzip -d GCF_000005845.2_ASM584v2_genomic.fna.gz &amp;&amp; cd ../..

</code></pre>
<p>You can download the ERCC fasta file that will be necessary for the normalization</p>
<pre><code class="language-bash">
cd input/data_folder &amp;&amp;
wget https://assets.thermofisher.com/TFS-Assets/LSG/manuals/ERCC92.zip &amp;&amp;
unzip ERCC92.zip &amp;&amp; rm ERCC92.zip &amp;&amp; rm ERCC92.gtf &amp;&amp; cd ../..

</code></pre>
<p>The annotation file can be download from <a href="https://github.com/foerstner-lab/GRADitude/blob/main/tutorial_data/NC_000913.3_no_duplicates.gff">tutorial_data folder</a> in github  (Höer <em>et</em> <em>al</em>., 2020, NAR)</p>
<pre><code class="language-bash">
cd input/data_folder &amp;&amp;
wget https://github.com/foerstner-lab/GRADitude/raw/main/tutorial_data/NC_000913.3_no_duplicates.gff &amp;&amp; cd ../..

</code></pre>
<p>Once all the raw reads are downloaded. We can start with removing 
the adapter using <a href="https://doi.org/10.14806/ej.17.1.200">cutadapt</a>.  </p>
<p>Before doing that we create the READemption folder where we will store the output that comes out from executing cutdadapt</p>
<pre><code class="language-bash">
singularity exec grad-seq-analysis-ecoli.sif reademption create -f READemption_analysis

</code></pre>
<p>If the commands works from your shell it should appear the READemption logo with no errors</p>
<pre><code class="language-bash">
   ___  _______   ___                 __  _
  / _ \/ __/ _ | / _ \___ __ _  ___  / /_(_)__  ___
 / , _/ _// __ |/ // / -_)  ' \/ _ \/ __/ / _ \/ _ \
/_/|_/___/_/ |_/____/\__/_/_/_/ .__/\__/_/\___/_//_/
                             / /
====================================================
========================================
=======================
==============

</code></pre>
<p>Once we create the READemption folder we can execute cutadapt.</p>
<pre><code class="language-bash">ls input/raw_reads | parallel -k -j 10 \
singularity exec grad-seq-analysis-ecoli.sif cutadapt \
-q 20 \
-m 1 \
-a &quot;AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC&quot; \
-o READemption_analysis/input/reads/'$(basename {})'.bz2 \
input/raw_reads/{} \
&gt; output/cutadapt_stats.txt

</code></pre>
<p>After executing cutadapt the trimmed output are already saved in the READemption/input/reads folder
as requested by READemption subcommand.</p>
<p>Now we have to copy the reference genome and the annotation 
in READemption/input/reference_sequences and READemption/input/annotations respectively
You can do it manually or you can run the following code:</p>
<pre><code class="language-bash">cp -r input/data_folder/NC_000913.3_no_duplicates.gff READemption_analysis/input/annotations/

</code></pre>
<pre><code class="language-bash">cp -r input/data_folder/GCF_000005845.2_ASM584v2_genomic.fna READemption_analysis/input/reference_sequences/

</code></pre>
<pre><code class="language-bash">cp -r input/data_folder/ERCC92.fa READemption_analysis/input/reference_sequences/

</code></pre>
<p>Now we can finally start with the alignment.
The -p parameter can be increase or decreased based on the CPUs that are available</p>
<pre><code class="language-bash">
singularity exec grad-seq-analysis-ecoli.sif reademption align \
--project_path READemption_analysis \
-p 4 \
-a 95 \
-l 20 \
--fastq \
--progress

</code></pre>
<p>After the alignment is finished the user can continue running the coverage subcommand </p>
<pre><code class="language-bash">
singularity exec grad-seq-analysis-ecoli.sif reademption coverage \
-p 4 \
--project_path READemption_analysis
</code></pre>
<p>Now the user can run the gene quantification. This subcommand will have as output one of the tables necessary for our further 
analysis with GRADitude.
With the parameter --features you can specify which features you want to quantify. For our analysis we decided to use 
CDS, rRNA, tRNA and ncRNA but many more features can be added, depending on the annotation file.</p>
<pre><code class="language-bash">    singularity exec grad-seq-analysis-ecoli.sif reademption gene_quanti \
        -p 4 \
        --features CDS,rRNA,tRNA,ncRNA \
        -l \
        --project_path READemption_analysis

</code></pre>
<p>After running the gene quantification, the 2 tables necessary for the downstream analysis are found in:</p>
<pre><code class="language-bash">READemption_analysis/output/align/reports_and_stats ---&gt; read_alignment_stats.csv
&amp;
READemption_analysis/output/gene_quanti/gene_quanti_combined --&gt; gene_wise_quantifications_combined.csv

</code></pre>
<p>Let's have a look at the folder structure we created right now:</p>
<pre><code class="language-bash">
├── grad-seq-analysis-ecoli.sif
├── input
└── output
└── READemption_analysis
    ├── input
    └── output


</code></pre>
<p>Now the user can create a folder GRADitude using the following subcommand</p>
<pre><code class="language-bash">
mkdir GRADitude &amp;&amp; mkdir GRADitude/input GRADitude/output

</code></pre>
<p>and then your folder structure will be like the one showed below where there is an input
and output folder inside the GRADitude folder created</p>
<pre><code class="language-bash">├── GRADitude
├── input
    └── output
├── grad-seq-analysis-ecoli.sif
├── input
└── output
└── READemption_analysis
    ├── input
    └── output
</code></pre>
<p>We now start the analysis copying all the necessary tables inside GRADitude</p>
<pre><code class="language-bash">  cp -r READemption_analysis/output/align/reports_and_stats/read_alignment_stats.csv ./GRADitude/input/ &amp;&amp;
  cp -r READemption_analysis/output/gene_quanti/gene_quanti_combined/gene_wise_quantifications_combined.csv ./GRADitude/input/
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../license/" class="btn btn-neutral float-right" title="License">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../perform_an_example_analysis/" class="btn btn-neutral" title="Performing an example analysis"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../perform_an_example_analysis/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../license/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
